pkg/kubelet/lifecycle/handlers_test.go

```diff
@@ -257,3 +257,460 @@ func TestRunHandlerHttpFailure(t *testing.T) {
 		t.Errorf("unexpected url: %s", fakeHTTPGetter.url)
 	}
 }
+
+func TestNoNewPrivsRequired(t *testing.T) {
+	admitPrivEsc := false
+	for _, testCase := range []struct {
+		name     string
+		input    *v1.Pod
+		expected bool
+	}{
+		{
+			input: &v1.Pod{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							SecurityContext: &v1.SecurityContext{
+								AllowPrivilegeEscalation: &admitPrivEsc,
+							},
+						},
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			input:    &v1.Pod{},
+			expected: false,
+		},
+	} {
+		res := noNewPrivsRequired(testCase.input)
+		if res != testCase.expected {
+			t.Errorf("expected admission result for pod to be %t, got %t", res, testCase.expected)
+		}
+	}
+}
+
+// NOTE: there _has_ to be something in kubernetes somewhere that does this already?
+func getBoolPointer(b bool) *bool {
+	return &b
+}
+
+func TestNoNewPrivsAdmitHandler(t *testing.T) {
```

Comment:
Ok, this test (and all the others) is nicely laid out, is easy to read and looks good. Could you please provide, in the commit message, as code comments o as github comment a pointer to understand the context and to learn about the expected behaviour? TL;DR: please add a reference so we can validate the tests themselves and understand the correct behaviour.

Reply:
i've addressed this in the second commit message.